tasks/cops_documentation.rake

```diff
@@ -138,16 +138,38 @@ task generate_cops_documentation: :yard_for_generate_documentation do
   end
   # rubocop:enable Metrics/MethodLength
 
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
   def references(config, cop)
     cop_config = config.for_cop(cop)
-    urls = RuboCop::Cop::MessageAnnotator.new(config, cop_config, {}).urls
+
+    # Since Rubocop v0.75.0 and above, the initializer for MessageAnnotator
+    # changed from:
+    #   def initialize(config, cop_config, options)
+    # to:
+    #   def initialize(config, cop_name, cop_config, options)
+    #
+    # Since this library has a loose Rubocop dependency, we select the
+    # right arguments based on the installed version.
+    rubocop_version = Gem::Version.new(RuboCop::Version::STRING)
+    urls =
+      if rubocop_version >= Gem::Version.new('0.75.0')
+        RuboCop::Cop::MessageAnnotator.new(
+          config, cop.name, cop_config, {}
+        ).urls
+      else
+        RuboCop::Cop::MessageAnnotator.new(config, cop_config, {}).urls
+      end
```

Comment:
WDYT of extracting this to a method, e.g.:
```ruby
    def references(config, cop)
      cop_config = config.for_cop(cop)
      urls = message_annotator.urls
      # ...
    end

    # Since RuboCop v0.75.0 ...
    def method_annotator
      rubocop_version = Gem::Version.new(RuboCop::Version::STRING)
      if rubocop_version >= Gem::Version.new('0.75.0')
        RuboCop::Cop::MessageAnnotator.new(config, cop.name, cop_config, {})
      else
        RuboCop::Cop::MessageAnnotator.new(config, cop_config, {})
      end
    end
```

I suspect this way both offences will be fixed.

Reply:
Can do, but `method_annotator` needs to take `config` and `cop_config` as arguments so it gets a bit verbose. Your call!

Reply:
Could we make a local version of `MessageAnnotator` instead? I’m thinking something like this:
```rb
# Since RuboCop v0.75.0 ...
rubocop_version = Gem::Version.new(RuboCop::Version::STRING)
if rubocop_version >= Gem::Version.new('0.75.0')
  MessageAnnotator = RuboCop::Cop::MessageAnnotator
else
  MessageAnnotator = Class.new(RuboCop::Cop::MessageAnnotator) do
    def initialize(config, cop_name, cop_config, options)
      super(config, cop_config, options)
    end
  end
end
```

Reply:
Started to do that, I think it looked neater as originally suggested by just extracting to a method 😄